substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrNativeEventWriter.java

```diff
@@ -0,0 +1,356 @@
+    }
+
+    @Uninterruptible(reason = "Accesses a native JFR buffer.", calleeMustBe = true)
+    public static void putChar(JfrNativeEventWriterData data, char v) {
+        if (ensureSize(data, Character.BYTES)) {
+            putUncheckedLong(data, v);
+        }
+    }
+
+    @Uninterruptible(reason = "Accesses a native JFR buffer.", calleeMustBe = true)
+    public static void putShort(JfrNativeEventWriterData data, short v) {
+        if (ensureSize(data, Short.BYTES)) {
+            putUncheckedLong(data, v & 0xFFFF);
+        }
+    }
+
+    @Uninterruptible(reason = "Accesses a native JFR buffer.", calleeMustBe = true)
+    public static void putInt(JfrNativeEventWriterData data, int v) {
+        if (ensureSize(data, Integer.BYTES)) {
+            putUncheckedLong(data, v & 0x00000000ffffffffL);
+        }
+    }
+
+    @Uninterruptible(reason = "Accesses a native JFR buffer.", calleeMustBe = true)
+    public static void putLong(JfrNativeEventWriterData data, long v) {
+        if (ensureSize(data, Long.BYTES)) {
+            putUncheckedLong(data, v);
+        }
+    }
+
+    @Uninterruptible(reason = "Accesses a native JFR buffer.", calleeMustBe = true)
+    public static void putUtf8(JfrNativeEventWriterData data, String string) {
+        int size = UninterruptibleUtils.String.modifiedUtf8Length(string, true);
+        if (ensureSize(data, size)) {
+            UninterruptibleUtils.String.toModifiedUtf8(string, data.getCurrentPos(), data.getEndPos(), true);
+        }
+    }
+
+    @Uninterruptible(reason = "Accesses a native JFR buffer.", calleeMustBe = true)
+    public static void putEventThread(JfrNativeEventWriterData data) {
+        putThread(data, CurrentIsolate.getCurrentThread());
+    }
+
+    @Uninterruptible(reason = "Accesses a native JFR buffer.", calleeMustBe = true)
+    public static void putThread(JfrNativeEventWriterData data, IsolateThread isolateThread) {
+        if (isolateThread.isNull()) {
+            putLong(data, 0L);
+        } else {
+            long threadId = SubstrateJVM.get().getThreadId(isolateThread);
+            putLong(data, threadId);
+        }
+    }
+
+    @Uninterruptible(reason = "Accesses a native JFR buffer.", calleeMustBe = true)
+    public static void putClass(JfrNativeEventWriterData data, Class<?> aClass) {
+        if (aClass == null) {
+            putLong(data, 0L);
+        } else {
+            putLong(data, SubstrateJVM.get().getClassId(aClass));
+        }
+    }
+
+    @Uninterruptible(reason = "Accesses a native JFR buffer.", calleeMustBe = true)
+    private static boolean ensureSize(JfrNativeEventWriterData data, int requested) {
+        assert requested > 0;
+        if (!isValid(data)) {
+            return false;
+        }
+
+        int totalRequested = requested + SIZE_SAFETY_CUSHION;
+        if (getAvailableSize(data).belowThan(totalRequested)) {
+            if (!accommodate(data, getUsedSize(data), totalRequested)) {
+                assert !isValid(data);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Use of identifier that is a keyword in later versions of Java (NM_FUTURE_KEYWORD_USED_AS_IDENTIFIER)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-use-of-identifier-that-is-a-keyword-in-later-versions-of-java-nm-future-keyword-used-as-identifier).
The identifier is a word that is reserved as a keyword in later versions of Java, and your code will need to be changed in order to compile it in later versions of Java.